util/ck_errf.pl: add functionality that brings it closer to util/mkerr.pl

levitte opened the pull request:
There was no option to give other config files than the default
crypto/err/openssl.ec, and yet it tried to check the errors generated
in engines (and failing, of course).

Also added the same '-internal' option as util/mkerr.pl.

Use these options in Configurations/unix-Makefile.tmpl

levitte committed:
util/ck_errf.pl: add functionality that brings it closer to util/mkerr.pl

There was no option to give other config files than the default
crypto/err/openssl.ec, and yet it tried to check the errors generated
in engines (and failing, of course).

Also added the same '-internal' option as util/mkerr.pl.

levitte committed:
make errors: use the new util/ck_errf.pl options

levitte committed:
.travis.yml: give make update exit code meaning again

levitte commented:
This also fixes a fault in our .travis.yml

levitte commented:
@DDvO, this fixes the issues you found in #6455

levitte committed:
Make ck_errf.pl ignore commented out error generation

DDvO commented:
Calling `util/ck_errf.pl -strict */*.c */*/*.c` yiels for me:
```
Use of uninitialized value $config in open at util/ck_errf.pl line 86.
Use of uninitialized value $config in concatenation (.) or string at util/ck_errf.pl line 86.
Trying to read : No such file or directory
```
The config parameter appears to be set when using `-internal`, but doing so I get:
```
Extra parameters given.
```
After patching the script adding a default value for `$config` like this:
```
my $config = "crypto/err/openssl.ec";
```
it works again (yet I do not see an improvement of the output style commented on in #6455.)

levitte committed:
fixup! util/ck_errf.pl: add functionality that brings it closer to util/mkerr.pl

levitte commented:
`-internal` sets the sources to check internally, just like `util/mkerr.pl` does.  So if you use that option, don't give any extra arguments.

Other than that, I have fixed the script so it *requires* `-conf` to be used if `-internal` isn't, and modified the output.

DDvO commented:
The issues I reported are now fixed :+1: 

Yet one more: after re-configuring, I tried `make errors` and got:
```
( b=`pwd`; set -e; cd .; \
  /usr/bin/perl util/ck_errf.pl -strict -internal; \
          /usr/bin/perl -I$b util/mkerr.pl  -internal )
( b=`pwd`; set -e; cd ./engines; \
          for E in *.ec ; do \
      /usr/bin/perl util/ck_errf.pl -strict -conf $E `basename $E .ec`.c; \
              /usr/bin/perl -I$b ../util/mkerr.pl  -static \
                -conf $E `basename $E .ec`.c ; \
          done )
Can't open perl script "util/ck_errf.pl": No such file or directory
make: *** [Makefile:566: errors] Error 1
```

levitte committed:
fixup! make errors: use the new util/ck_errf.pl options

levitte commented:
Fixed!

t-j-h reviewed (approved):
I've been eyeballing the changes ...

DDvO commented:
Fix confirmed

levitte commented:
Merged.

3ccfcd8a851b49c7e4b1e40bb1d7ccd326e6f94f Make ck_errf.pl ignore commented out error generation
caf12bd6754987d99b9e27f2422136a48e53cc86 .travis.yml: give make update exit code meaning again
aa6cc8d33c107c8b6d2e9bbffe52086e191f607b make errors: use the new util/ck_errf.pl options
b9def672b123ce3d1102683f13c61bc68a07c5b0 util/ck_errf.pl: add functionality that brings it closer to util/mkerr.pl
